### PR TITLE
🧪 Add tests for LLM Error Normalization

### PR DESCRIPTION
**Pull request by arn0ld87**
🎯 **What:** Added tests for the `normalize_provider_error` and `LlmProviderError` logic in `backend/app/llm/errors.py`.
📊 **Coverage:** Covered all OpenAI specific errors (Timeout, Connection, RateLimit, ServerError, ClientError) as well as HTTPX errors (Timeout, Connect). Added tests for edge cases like extracting status code from response and checking fallback unknown errors.
✨ **Result:** 100% test coverage for `backend/app/llm/errors.py`.

---
*PR created automatically by Jules for task [8719323655096232548](https://jules.google.com/task/8719323655096232548) started by @arn0ld87*

### CODE REVIEW

File: backend/tests/llm/test_errors.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        self.status_code = status_code
+        self.request = DummyRequest()
+        self.headers = {}
+
+class DummyExceptionWithStatus(Exception):
+    def __init__(self, status_code):
+        self.status_code = status_code
+        super().__init__("DummyExceptionWithStatus")
+
+def test_normalize_openai_timeout():
```
[the remark this answers is not in the document]
Comment (arn0ld87) [reply backend/tests/llm/test_errors.py:26]: Behoben mit `895fcfb5`: `docs/STATUS.md` wurde via `scripts/sync-status.sh` regeneriert (Backend Tests collected: 3903 → 3915). `bash scripts/sync-status.sh --check` lokal verifiziert — kein Drift mehr, der `backend-pr-gate` sollte damit wieder grün sein.
